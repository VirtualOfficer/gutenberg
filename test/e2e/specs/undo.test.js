/**
 * Internal dependencies
 */
import {
	newPost,
	pressWithModifier,
	getEditedPostContent,
} from '../support/utils';

describe( 'undo', () => {
	beforeAll( async () => {
		await newPost();
	} );

	it( 'Should undo to expected level intervals', async () => {
		await page.click( '.editor-default-block-appender__content' );

		await page.keyboard.type( 'This' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'is' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'test' );

		expect( await getEditedPostContent() ).toMatchSnapshot();

		await pressWithModifier( 'mod', 'z' ); // Strip 3rd paragraph text
		await pressWithModifier( 'mod', 'z' ); // Strip 3rd paragraph block
		await pressWithModifier( 'mod', 'z' ); // Strip 2nd paragraph text
		await pressWithModifier( 'mod', 'z' ); // Strip 2nd paragraph block
		await pressWithModifier( 'mod', 'z' ); // Strip 1st paragraph text
		await pressWithModifier( 'mod', 'z' ); // Strip 1st paragraph block

		// Should have no more history.
		await page.waitForSelector( '.editor-history__undo:disabled' );

		expect( await getEditedPostContent() ).toBe( '' );
	} );
} );
