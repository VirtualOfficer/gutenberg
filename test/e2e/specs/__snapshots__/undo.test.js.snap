// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undo Should undo to expected level intervals 1`] = `
"<!-- wp:paragraph -->
<p>This</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>is</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>test</p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after arrow navigation 1`] = `
"<!-- wp:paragraph -->
<p>before keyboar after keyboardd</p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after arrow navigation 2`] = `
"<!-- wp:paragraph -->
<p>before keyboard</p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after mouse move 1`] = `
"<!-- wp:paragraph -->
<p>before move after move</p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after mouse move 2`] = `
"<!-- wp:paragraph -->
<p>before move</p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after non input change 1`] = `
"<!-- wp:paragraph -->
<p>before keyboard <strong>after keyboard</strong></p>
<!-- /wp:paragraph -->"
`;

exports[`undo should undo typing after non input change 2`] = `
"<!-- wp:paragraph -->
<p>before keyboard </p>
<!-- /wp:paragraph -->"
`;
